utils/filter.c

```diff
@@ -940,6 +991,11 @@ int setup_trigger_action(char *str, struct uftrace_trigger *tr, char **module,
 
 			break;
 		}
+		if (!strncasecmp(pos, "clear", 5)) {
+			/* preserve orig flags overwritten by 'clear parser' */
+			tr->flags |= orig_flags;
```

Comment:
I don't want to touch `tr->flags` here.  How about this and move it out of the loop?

```c
strv_for_each(&acts, pos, j) {
    ...
}
if (tr->flags & TRIGGER_FL_CLEAR) {
    if (orig_flags && tr->clear_flags == ~0)
        tr->clear_flags = orig_flags;
}
```

Reply:
Yeah, I think that's reasonable.  There's no point to use clear for the same action, it seems like a user error.  But anyway, the clear action is handled first and then we add the new flags.  So it should be fine.